include/tvm/relax/op_attr_types.h

```diff
@@ -49,6 +49,16 @@ using FInferStructInfo =
  */
 using FCallPacked = String;
 
+/*!
+ * \brief Gradient for a specific op.
+ *
+ * \param orig_call the original Expr.
+ * \param output_grad the gradient of the Expr.
+ * \return the gradient for each parameter.
+ */
+using FPrimalGradient = runtime::TypedPackedFunc<tvm::Array<Expr>(
+    const Var& orig_var, const Call& orig_call, const Var& output_grad, BlockBuilder ctx)>;
```

Comment:
Maybe `const BlockBuilder&`?

Reply:
I'm not sure since we may use `ctx.emit`. Is it still const?